ipc/provider/src/manager/evm/manager.rs

```diff
@@ -136,16 +169,27 @@ impl TopDownCheckpointQuery for EthSubnetManager {
 
         let ev = contract
             .event::<NewStakingRequest>()
-            .from_block(start as u64)
-            .to_block(end as u64);
-        let mut event_stream = ev.stream().await?;
+            .from_block(epoch as u64)
+            .to_block(epoch as u64);
+        let mut event_stream = ev.stream_with_meta().await?;
 
         let mut changes = vec![];
-        while let Some(Ok(event)) = event_stream.next().await {
+        let mut hash = None;
+        while let Some(Ok((event, meta))) = event_stream.next().await {
+            if let Some(h) = hash {
+                if h != meta.block_hash {
+                    return Err(anyhow!("block hash not equal"));
+                }
+            } else {
+                hash = Some(meta.block_hash);
+            }
             changes.push(StakingChangeRequest::try_from(event)?);
         }
 
-        Ok(changes)
+        Ok(TopDownQueryPayload {
+            value: changes,
+            block_hash: hash.unwrap().0.to_vec(),
```

Comment:
What if there were no events at that block height? It looks to me like `hash` would stay empty and this `.unwrap()` would panic.

Reply:
yep, you are right. In this case, I should have changed the return type to: `Option<TopDownQueryPayload<Vec<StakingChangeRequest>>>`